### DISCUSSION TITLE
SSL certificate error after mcservice start is run

### DISCUSSION THREAD

**Discussion by girishmalar** [nextgenhealthcare/connect#6385] [Q&A] [answered]
I am a newbie to mirth stuff but I am looking at some errors I get when I try to do a mcservice start.
The server never appears to start and I see this error in the logs which seems related to trusting some certificate

transformer: JavaException: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

Any pointers that I could use to get past this?

thanks so much

**Comment by pacmano1** [comment]
Do you license the Core Pack from Nextgen?  Which has the TLS extension?  

If not, you need to add the certificate chain to the operating system's java install - there are lots of posts already on this problem.

**Reply by girishmalar** [reply]
Thanks a lot, I updated the root certs by copying the relevant root certs and doing a update-ca-certificates —fresh. 
However the mirth doesn't seem to pick it up.. even after a restart. can you point me to a specific post that answers how to add the cert chain?

**Reply by pacmano1** [reply, marked as answer]
I use keytool or KeyStore explorer to import the CA the cacerts file.  On Ubuntu, the file is `/etc/ssl/certs/java/cacerts`
